frontend/packages/integration-tests-cypress/tests/crud/k8-openshift-cruds.cyspec.ts

```diff
@@ -0,0 +1,190 @@
+    .set('pods', { kind: 'Pod' })
+    .set('services', { kind: 'Service' })
+    .set('serviceaccounts', { kind: 'ServiceAccount' })
+    .set('secrets', { kind: 'Secret' })
+    .set('configmaps', { kind: 'ConfigMap' })
+    .set('persistentvolumes', { kind: 'PersistentVolume', namespaced: false })
+    .set('storageclasses', { kind: 'StorageClass', namespaced: false })
+    .set('ingresses', { kind: 'Ingress' })
+    .set('cronjobs', { kind: 'CronJob' })
+    .set('jobs', { kind: 'Job' })
+    .set('daemonsets', { kind: 'DaemonSet' })
+    .set('deployments', { kind: 'Deployment' })
+    .set('replicasets', { kind: 'ReplicaSet' })
+    .set('replicationcontrollers', { kind: 'ReplicationController' })
+    .set('persistentvolumeclaims', { kind: 'PersistentVolumeClaim' })
+    .set('statefulsets', { kind: 'StatefulSet' })
+    .set('resourcequotas', { kind: 'ResourceQuota' })
+    .set('limitranges', { kind: 'LimitRange' })
+    .set('horizontalpodautoscalers', { kind: 'HorizontalPodAutoscaler' })
+    .set('networkpolicies', { kind: 'NetworkPolicy' })
+    .set('roles', { kind: 'Role' });
+  const openshiftObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>()
+    .set('deploymentconfigs', { kind: 'DeploymentConfig' })
+    .set('buildconfigs', { kind: 'BuildConfig' })
+    .set('imagestreams', { kind: 'ImageStream' })
+    .set('routes', { kind: 'Route' })
+    .set('user.openshift.io~v1~Group', { kind: 'user.openshift.io~v1~Group', namespaced: false });
+  const serviceCatalogObjs = OrderedMap<string, { kind: string; namespaced?: boolean }>().set(
+    'clusterservicebrokers',
+    {
+      kind: 'servicecatalog.k8s.io~v1beta1~ClusterServiceBroker',
+      namespaced: false,
+    },
+  );
+
+  let testObjs = Cypress.env('openshift') === true ? k8sObjs.merge(openshiftObjs) : k8sObjs;
+  testObjs = Cypress.env('servicecatalog') === true ? testObjs.merge(serviceCatalogObjs) : testObjs;
+  const testLabel = 'automatedTestName';
+  const resourcesWithCreationForm = new Set(['StorageClass', 'Route', 'PersistentVolumeClaim']);
+  const leakedResources = new Set<string>();
+
+  testObjs.forEach(({ kind, namespaced = true }, resource) => {
+    describe(kind, () => {
+      const name = `${testName}-${_.kebabCase(kind)}`; // end name with '&' to force test failures
+      it('displays a list view for the resource', () => {
+        cy.visit(
+          `${namespaced ? `/k8s/ns/${testName}` : '/k8s/cluster'}/${resource}?name=${testName}`,
+        );
+      });
+      if (namespaced) {
+        it('has a working namespace dropdown on namespaced objects', () => {
+          listPage.projectDropdownShouldExist();
+          listPage.projectDropdownShouldContain(testName);
+        });
+      } else {
+        it('does not have a namespace dropdown on non-namespaced objects', () => {
+          listPage.projectDropdownShouldNotExist();
+        });
+      }
+      it('creates, searches for, edits, and deletes a resource instance', () => {
```

Comment:
This test is really long. Can we create the resource in `before()` and delete the resource in `after()`? Then we can break these up into separate tests and still have test independence.

Reply:
yeah, that seems to be what happens when we strive to make them all independent of each other.

Reply:
But it seems pretty easy to fix. Just create the resource in `before`, delete it in `after`, and make sure there is a `cy.visit` at the top of each test.